code/modules/mob/living/carbon/xenomorph/XenoProcs.dm

```diff
@@ -216,16 +216,45 @@
 	armor_integrity = min(armor_integrity + value, 100)
 
 
-//Strip all inherent xeno verbs from your caste. Used in evolution.
+///Strip all inherent xeno verbs from your caste. Used in evolution.
 /mob/living/carbon/xenomorph/proc/remove_inherent_verbs()
 	if(inherent_verbs)
 		remove_verb(src, inherent_verbs)
 
-//Add all your inherent caste verbs and procs. Used in evolution.
+///Add all your inherent caste verbs and procs. Used in evolution.
 /mob/living/carbon/xenomorph/proc/add_inherent_verbs()
 	if(inherent_verbs)
 		add_verb(src, inherent_verbs)
 
+/// activates the buffs and debuffs of Stalk mode for certain castes , This is the generic proc but each xeno can have their own set of snowflake.
+/mob/living/carbon/xenomorph/set_movement_intent(new_intent, cooldown = 3 SECONDS)
+	if(TIMER_COOLDOWN_CHECK(src, STALKING_COOLDOWN))
+		to_chat(src, SPAN_XENOWARNING("We are not ready to change our stance"))
+		return
+	TIMER_COOLDOWN_START(src, STALKING_COOLDOWN, cooldown)
+	. = ..()
+	if(caste)
+		walk_modifier = caste.walk_modifier
+		run_modifier = caste.run_modifier
+	if(!can_ventcrawl()) // for now only crawlers can get actual buffs for stalk mode
+		return
+	if(tier > 1)
+		return
+	switch(new_intent)
+		if(MOVE_INTENT_HUNT)
+			evasion_modifier = XENO_EVASION_NONE
+			remove_temp_pass_flags(PASS_MOB_IS_XENO|PASS_TYPE_CRAWLER|PASS_MOB_THRU_XENO)
+			remove_temp_can_pass_flags(PASS_MOB_IS_XENO|PASS_MOB_THRU_XENO)
+			layer = MOB_LAYER
+
+		if(MOVE_INTENT_STALK)
+			evasion_modifier += XENO_EVASION_MOD_ULTRA // this looks like a lot but evasion is broken and doesnt even work most of the time
+			add_temp_pass_flags(PASS_MOB_IS_XENO|PASS_MOB_THRU_XENO|PASS_TYPE_CRAWLER)
+			add_temp_can_pass_flags(PASS_MOB_IS_XENO|PASS_MOB_THRU_XENO)
+			layer = BELOW_TABLE_LAYER
```

Comment:
I'd rather we just not fight with the layers, but if you do want to alter the layer then there's a bunch of things we need to do:
- whenever you want to restore the layer you need to do it via `initial(layer)` instead
- changing intent to stalk needs to only set the layer if its not currently `XENO_HIDING_LAYER`
- changing intent to hunt needs to only reset the layer if its currently on `BELOW_TABLE_LAYER` (or whatever layer is set when stalking)
- unhide needs to look at the move intent and set the layer back to `BELOW_TABLE_LAYER` (or whatever layer is set when stalking) if stalking
- shimmy component I will need to make similar exceptions for hybrisa